src/com/lftechnology/java/training/srijan/basics/coreapp/controller/EmployeeManagementController.java

```diff
@@ -0,0 +1,307 @@
+			int terminateEmployee = editAndTerminateEmployee();
+			employeeServiceImpl.terminate(terminateEmployee);
+			break;
+		case 4:
+			employees = employeeServiceImpl.viewAllUser(e);
+			LOGGER.log(Level.INFO, "{0}", employees);
+			break;
+		default:
+			LOGGER.log(Level.INFO, "Wrong option, Default view functionality");
+			employees = employeeServiceImpl.viewAllUser(e);
+			LOGGER.log(Level.INFO, "{0}", employees);
+			break;
+		}
+	}
+
+	/**
+	 * <p>Takes employee information from user</p>
+	 * 
+	 * @param e
+	 * @return
+	 * @author srijan
+	 */
+	public Employee inputUserInfo(Employee e) {
+
+		Scanner input = null;
+		try {
+			input = new Scanner(System.in);
+			LOGGER.log(Level.INFO, "Enter Username*:");
+			this.username = input.nextLine();
+			e.setUserName(username);
+			LOGGER.log(Level.INFO, "Enter Fullname*:");
+			this.fullname = input.nextLine();
+			e.setFullName(fullname);
+			LOGGER.log(Level.INFO, "Enter Department*:");
+			this.department = input.nextLine();
+			e.setDepartment(department);
+			LOGGER.log(Level.INFO, "Enter Address:");
+			this.address = input.nextLine();
+			e.setAddress(address);
+			LOGGER.log(Level.INFO, "Enter password*");
+			this.password = input.nextLine();
+			e.setPassword(password);
+			LOGGER.log(
+				Level.INFO,
+				"Enter User Role*:\n1.ADMIN\n2.NORMAL\n3.Default: NORMAL");
+			this.choiceUserRole = input.nextInt();
+			switch (choiceUserRole) {
+			case 1:
+				e.setUserRole(UserRole.ADMIN);
+				break;
+			case 2:
+				e.setUserRole(UserRole.NORMAL);
+				break;
+			default:
+				LOGGER.log(
+					Level.INFO,
+					"please choose right option,Default Normal user assigned");
+				e.setUserRole(UserRole.NORMAL);
+				break;
+			}
+			LOGGER.log(
+				Level.INFO,
+				"User status:\n0.Not terminated\n1.Terminated\n2.Default: Not Terminated");
+			userStatus = input.nextInt();
+			switch (userStatus) {
+			case 0:
+				e.setIsTerminated(false);
+				break;
+			case 1:
+				e.setIsTerminated(true);
+				break;
+			default:
+				LOGGER.log(
+					Level.INFO,
+					"Wrong option, User has been default assigned to Not Terminated");
+				e.setIsTerminated(false);
+				break;
+			}
+		}
+		catch (Exception ex) {
+			LOGGER.log(Level.INFO, "Exception", ex);
+		}
+		finally {
+			input.close();
+		}
+		return e;
+	}
+
+	/**
+	 * <p>Calls different user methods depending upon user choice</p>
+	 * 
+	 * @param normalUserChoice
+	 * @param userName
+	 * @author srijan
+	 */
+	public void normalUserOptionHandler(int normalUserChoice, String userName) {
+
+		switch (normalUserChoice) {
+		case 1:
+			employees = employeeServiceImpl.viewAllUser(e);
+			LOGGER.log(Level.INFO, "{0}", employees);
+			break;
+		case 2:
+			employee = employeeServiceImpl.findUserByUserName(userName);
+			LOGGER.log(Level.INFO, "{0}", employee);
+			Employee changedProfileDetails = changeProfile(employee);
+			employeeServiceImpl.updateProfile(changedProfileDetails);
+			break;
+		default:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1151)